### DISCUSSION TITLE
Option

### DISCUSSION THREAD

**Discussion by steverawlins-zebra** [zalopay-oss/jmeter-grpc-request#121] [Ideas] [answered]
We have discovered, during high-volume testing (more than 1000 messages per second), CHANNEL SHUTDOWN problems within jmeter-grpc-request.  Furthermore, once a CHANNEL SHUTDOWN triggers, the remainder of the test is useless - when we hit a problem, it's "game over" and what follows is nothing but a stream of CHANNEL SHUTDOWNs.

Please add a configuration option allowing us to increase that channel-shutdown timeout from the hard-coded 5, to a larger number such as 60 (seconds)

Thank you

P.S. The basis for our idea comes from examining the code in    `jmeter-grpc-request/src/main/java/vn/zalopay/benchmark/core`

```
 public void shutdownNettyChannel() {
        try {
            if (channel != null) {
                channel.shutdown();
                channel.awaitTermination(5, TimeUnit.SECONDS);
            }
        } catch (InterruptedException e) {
            throw new RuntimeException("Caught exception while shutting down channel", e);
        }
    }
```

**Comment by minhhoangvn** [comment]
Can I know do you use the for-loop with the gRPC sampler, currently, this issue maybe happen if we use the for-loop with the gRPC sampler, so, when the channel is shut down, it can't send a request anymore. Please also provide your sample JMeter test plan to reproduce that issue for debugging purpose :D

**Comment by minhhoangvn** [comment, marked as answer]
We have updated plugin, allowing users can config channel shutdown time 
https://github.com/zalopay-oss/jmeter-grpc-request/pull/124

**Comment by steverawlins-zebra** [comment]
Thank you very much.  We have downloaded v1.2.2 for evaluation.
